quinn/src/endpoint.rs

```diff
@@ -91,23 +91,14 @@ impl Endpoint {
         runtime: Arc<dyn Runtime>,
     ) -> io::Result<Self> {
         let socket = runtime.wrap_udp_socket(socket)?;
-        Self::new_with_runtime(config, server_config, socket, runtime)
+        Self::new_with_abstract_socket(config, server_config, socket, runtime)
     }
 
     /// Construct an endpoint with arbitrary configuration and pre-constructed abstract socket
     ///
     /// Useful when `socket` has additional state (e.g. sidechannels) attached for which shared
     /// ownership is needed.
-    pub fn new_with_abstract_socket(
-        config: EndpointConfig,
-        server_config: Option<ServerConfig>,
-        socket: Box<dyn AsyncUdpSocket>,
-        runtime: Arc<dyn Runtime>,
-    ) -> io::Result<Self> {
-        Self::new_with_runtime(config, server_config, socket, runtime)
-    }
-
-    fn new_with_runtime(
+    fn new_with_abstract_socket(
```

Comment:
Um, this replaced the public constructor with only a private one. That doesn't seem like the right direction?

Reply:
Ops. Sorry. https://github.com/quinn-rs/quinn/pull/1596

Reply:
To be fair, it's also @Ralith's bad for not catching it in review! Thanks for following up.